lightning/src/ln/onchaintx.rs

```diff
@@ -0,0 +1,696 @@
+		let destination_script = Readable::read(reader)?;
+
+		let pending_claim_requests_len: u64 = Readable::read(reader)?;
+		let mut pending_claim_requests = HashMap::with_capacity(cmp::min(pending_claim_requests_len as usize, MAX_ALLOC_SIZE / 128));
+		for _ in 0..pending_claim_requests_len {
+			pending_claim_requests.insert(Readable::read(reader)?, Readable::read(reader)?);
+		}
+
+		let claimable_outpoints_len: u64 = Readable::read(reader)?;
+		let mut claimable_outpoints = HashMap::with_capacity(cmp::min(pending_claim_requests_len as usize, MAX_ALLOC_SIZE / 128));
+		for _ in 0..claimable_outpoints_len {
+			let outpoint = Readable::read(reader)?;
+			let ancestor_claim_txid = Readable::read(reader)?;
+			let height = Readable::read(reader)?;
+			claimable_outpoints.insert(outpoint, (ancestor_claim_txid, height));
+		}
+		let waiting_threshold_conf_len: u64 = Readable::read(reader)?;
+		let mut onchain_events_waiting_threshold_conf = HashMap::with_capacity(cmp::min(waiting_threshold_conf_len as usize, MAX_ALLOC_SIZE / 128));
+		for _ in 0..waiting_threshold_conf_len {
+			let height_target = Readable::read(reader)?;
+			let events_len: u64 = Readable::read(reader)?;
+			let mut events = Vec::with_capacity(cmp::min(events_len as usize, MAX_ALLOC_SIZE / 128));
+			for _ in 0..events_len {
+				let ev = match <u8 as Readable<R>>::read(reader)? {
+					0 => {
+						let claim_request = Readable::read(reader)?;
+						OnchainEvent::Claim {
+							claim_request
+						}
+					},
+					1 => {
+						let outpoint = Readable::read(reader)?;
+						let input_material = Readable::read(reader)?;
+						OnchainEvent::ContentiousOutpoint {
+							outpoint,
+							input_material
+						}
+					}
+					_ => return Err(DecodeError::InvalidValue),
+				};
+				events.push(ev);
+			}
+			onchain_events_waiting_threshold_conf.insert(height_target, events);
+		}
+
+		Ok(OnchainTxHandler {
+			destination_script,
+			claimable_outpoints,
+			pending_claim_requests,
+			onchain_events_waiting_threshold_conf,
+			secp_ctx: Secp256k1::new(),
+			logger,
+		})
+	}
+}
+
+impl OnchainTxHandler {
+	pub(super) fn new(destination_script: Script, logger: Arc<Logger>) -> Self {
+		OnchainTxHandler {
+			destination_script,
+			pending_claim_requests: HashMap::new(),
+			claimable_outpoints: HashMap::new(),
+			onchain_events_waiting_threshold_conf: HashMap::new(),
+
+			secp_ctx: Secp256k1::new(),
+			logger,
+		}
+	}
+
+	pub(super) fn get_witnesses_weight(inputs: &[InputDescriptors]) -> usize {
+		let mut tx_weight = 2; // count segwit flags
+		for inp in inputs {
+			// We use expected weight (and not actual) as signatures and time lock delays may vary
+			tx_weight +=  match inp {
+				// number_of_witness_elements + sig_length + revocation_sig + pubkey_length + revocationpubkey + witness_script_length + witness_script
+				&InputDescriptors::RevokedOfferedHTLC => {
+					1 + 1 + 73 + 1 + 33 + 1 + 133
```

Comment:
not for this PR, but I wonder if it will be more robust to create a sample tx and serialize it to get these numbers

Reply:
Sure further dry-up of this code with some tx template what somehow in my mind